### PR TITLE
feat: year in hog is back baby

### PR DESCRIPTION
**Pull request by pauldambra**
What the world needs now, as Burt Bacarach once said, is more products doing the spotify wrapped thing.

And he was right!

Let's resurrect Year! In! Hog!

## maybe to do

- [x] fixed year in hog button for PostHog 3000
- [ ] check the badge boundaries make sense for 2023 data
- [ ] add a survey badge

### CODE REVIEW

File: frontend/src/layout/navigation/TopBar/YearInHogButton.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                width={'36'}
+                height={'36'}
+                className={clsx(
+                    'CheekyHog',
+                    isHovering && 'CheekyHog--peek',
+                    isHovering === false && 'CheekyHog--hide'
+                )}
+            />
+            {/*need to sync background color with button*/}
+            <div className={'rounded absolute top-0 left-0 w-full h-full YearInHog__mask'} />
```
Comment (daibhin) [inline frontend/src/layout/navigation/TopBar/YearInHogButton.tsx:23]:
```suggestion
            <div className={'absolute top-0 left-0 w-full h-full YearInHog__mask'} />
```

related to comment above

File: posthog/year_in_posthog/sharing-buttons.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                  d="M19 3A2 2 0 0 1 21 5V19A2 2 0 0 1 19 21H5A2 2 0 0 1 3 19V5A2 2 0 0 1 5 3H19M18.5 18.5V13.2A3.26 3.26 0 0 0 15.24 9.94C14.39 9.94 13.4 10.46 12.92 11.24V10.13H10.13V18.5H12.92V13.57C12.92 12.8 13.54 12.17 14.31 12.17A1.4 1.4 0 0 1 15.71 13.57V18.5H18.5M6.88 8.56A1.68 1.68 0 0 0 8.56 6.88C8.56 5.95 7.81 5.19 6.88 5.19A1.69 1.69 0 0 0 5.19 6.88C5.19 7.81 5.95 8.56 6.88 8.56M8.27 18.5V10.13H5.5V18.5H8.27Z"/>
+        </svg>
+    </a>
+</div>
+<div>
+    <a target="_blank"
+       aria-label="share year in posthog 2022 to twitter"
+       data-attr="share-year-in-posthog-2022-to-twitter"
+       href="https://twitter.com/intent/tweet?text={{ 'No hedgehogs were harmed in the course of this year on @posthog' }}&url={{ page_url | urlencode }}&hashtags=YearInPostHog">
+        <svg viewBox="0 0 24 24">
```
Comment (daibhin) [inline posthog/year_in_posthog/sharing-buttons.html:29]: @elonmusk won't be happy

Screenshot 2023-12-06 at 14 07 39

**Review by daibhin — APPROVED** [review]
Ran it and everything looked great! I didn't spend much time reviewing the calculation because I assumed it was ported over from last year

File: posthog/year_in_posthog/sharing-buttons.html (hunk 2 above)
Comment (pauldambra) [reply posthog/year_in_posthog/sharing-buttons.html:29]: I only just noticed you tagged him... now we're in trouble 🤣

**Pull request merged**